lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/SQLTranslatorUtils.java

```diff
@@ -395,14 +396,21 @@ static void translateAllColumnReferences(Element root, TableAndColumnMapper mapp
 		if (ColumnType.DOUBLE.equals(match.getColumnTranslationReference().getColumnType())) {
 			if (columnReference.isInContext(SelectList.class)) {
 				if (!columnReference.isInContext(HasFunctionReturnType.class)) {
-					return Optional
-							.of(createDoubleExpanstion(mapper.getNumberOfTables(), match.getTableInfo().getTranslatedTableAlias(),
-									match.getColumnTranslationReference().getTranslatedColumnName()));
+					SqlContext context = columnReference.getContext(HasSqlContext.class).get().getSqlContext();

```

Comment:
Let us extract this into its own function and use an orElseThrow(IllegalStateException)